### ISSUE TITLE
Trigger helm chart updates from the UI

### ISSUE THREAD

**Issue by jeroenlandheer** [weaveworks/vscode-gitops-tools#432]
Maybe I'm missing something, but I can't seem to find a way to force Flux to update a Helm chart when I know there's a new version in my OCI repo. 

Once I've waited long enough that the timer runs out, it'll get installed.... but these timers here are a bit long. How do we trigger a manual update?  

This might be related to: https://github.com/fluxcd/flux2/issues/3656

**Comment by kingdonb** [comment]
💯 good research, that's the reason why we can't have a chart reconcile yet.

It is possible to set `spec.chart.spec.interval` if this is blocking you from using OCI in your development workflows. It is not ideal, but Helm should get a lot more attention once Flux lands a 2.0.0 release, and after KubeCon.

**Comment by jeroenlandheer** [comment]
It looks like there's another way to trigger an immediate reconcile:

```shell
kubectl annotate --field-manager=flux-client-side-apply --overwrite helmchart/<chart-name >reconcile.fluxcd.io/requestedAt="$(date +%s)" -n <chart-namespace>
```

You could use that under the hood to enable this feature.

**Comment by kingdonb** [comment]
Let's try to address this in the next release – if the Flux CLI can have the feature added, great, but if not we can use the kubectl annotate (thanks @jeroenlandheer for providing this)

**Comment by kingdonb** [comment]
The `flux reconcile source chart` command has been added in 2.0.x Flux branch, we didn't get around to this one yet.

First we need to make the chart visible, or add a handle to the right-click menu of each HelmRelease to reconcile the chart.

You might be able to use the new `Reconcile with Source` but I have a feeling that is only going to reconcile the HelmRepo and not the Chart source. @juozasg let's see if we can add this one to the prerelease branch before we start the 0.26 train.

**Comment by juozasg** [comment]
Yes we should visualize HelmChart. I think it wont make the UI too confusing to just list them in Workloads tree?

**Comment by kingdonb** [comment]
It will definitely be too confusing. IDK, we have to see what it looks like. Can we somehow list them as a sub-resource under HelmChart, next to all the templated manifests (but with the iconography of a source?)

I think a good experience would be: see red HelmRelease, hover mouse over. Error says "problem with HelmChart!" click the unfold triangle. See HelmChart is red, mouse over, see error message (whatever failure is found there in the conditions, we can use the same mechanism we use to show status from workloads and other sources...)

Right click HelmChart, reconcile, watch the release proceed immediately after that

It might also be nice to have "Reconcile with Chart" as a right-click option on the HelmRelease itself

**Comment by kingdonb** [comment]
I noticed that we are already reconciling the HelmChart on 0.25.0 when you click `Reconcile with Source` 💯 

I don't know if we want to change that at all, to make it more obvious that is what you're going to get, or change it so that the HelmChart is a bit more visible, I think that either of those changes would still be in scope for sure.

But JFYI @jeroenlandheer I think we've used this bit of information in 0.25.0, (https://github.com/weaveworks/vscode-gitops-tools/issues/432#issuecomment-1529372885) at least it seems that was done, and it seems to work based on my testing

The Flux CLI itself also has `flux reconcile source chart` now 👍 

Keeping this open for the UX enhancements we can follow on

**Comment by kingdonb** [comment]
Just checking in, we've had v0.25.1 release today with mega performance improvements.

It does not specifically address this issue, but I see we have sort of already addressed the issue. Not sure if it's completely solved, or if we should keep this open now. I need to replicate a Helm chart not ready condition, I guess the easiest way to do that would be to delete the source controller pod and set HelmRelease to immediately reconciling.

I will circle around and look at this one again before v0.25.2, but if you've tried the latest release and have any feedback you are most welcome to provide it.

**Comment by kingdonb** [comment]
We've had 0.26.0 and I think it is possible to trigger Helm chart reconciles from the UI now. I think it has been for some time in prerelease.

<img width="1431" alt="Screenshot 2024-01-15 at 9 09 56 AM" src="https://github.com/weaveworks/vscode-gitops-tools/assets/3286998/ce2998aa-3e21-4228-829b-80f1e39a96a4">
<img width="314" alt="Screenshot 2024-01-15 at 9 10 01 AM" src="https://github.com/weaveworks/vscode-gitops-tools/assets/3286998/8e4d8fd5-4b90-40b7-847d-3170c7de8748">

Please let us know if we can close this issue. 🙏

**Comment by kingdonb** [comment]
We can reopen it if needed. (Closing)
